### ISSUE TITLE
libreadline on OS X redux

### ISSUE THREAD

**Issue by sandiway** [SWI-Prolog/issues#46]
"SWI-Prolog is linked to the real GNU libreadline as the emulation does not support its requirements.
libreadline comes from Macports or Homebrew or whatever. Except for the binaries distributed on the SWI-Prolog page, I have little control of the version of libreadline used."

I'm sorry but I guess that's what I'm not sure I completely understand. otool -L reports that the binary swipl is NOT (dynamically) linked to libreadline so it's not like it's inadvertantly picking up a wrong version, see below. I guess whoever complied 7.3.x swipl on the mac must have used macports or homebrew that is not fully compatible with later versions of OS X. Can the person who compiled the binary available for download possibly chime in here? Thanks!

otool -L /Applications/SWI-Prolog.app/Contents/MacOS/swipl
/Applications/SWI-Prolog.app/Contents/MacOS/swipl:
	@executable_path/../swipl/lib/x86_64-darwin15.6.0/libswipl.dylib (compatibility version 0.0.0, current version 7.3.27)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)

**Comment by JanWielemaker** [comment]
```
MW-020723 (Contents) 508_> cd MacOS/
MW-020723 (MacOS) 509_> ls
swipl*		swipl-ld*	swipl-rc*	swipl-win*
MW-020723 (MacOS) 510_> otool -L swipl
swipl:
	@executable_path/../swipl/lib/x86_64-darwin15.6.0/libswipl.dylib (compatibility version 0.0.0, current version 7.3.27)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

Ok, but otool doesn't recurse, so we need to continue on the shared lib that is the actual SWI-Prolog
system to get this:

````
MW-020723 (MacOS) 511_> otool -L ../swipl/lib/x86_64-darwin15.6.0/libswipl.dylib../swipl/lib/x86_64-darwin15.6.0/libswipl.dylib:
	/Users/janw/lib/swipl/lib/x86_64-darwin15.6.0/libswipl.dylib (compatibility version 0.0.0, current version 7.3.27)
	@executable_path/../swipl/lib/x86_64-darwin15.6.0/libgmp.10.dylib (compatibility version 14.0.0, current version 14.0.0)
	@executable_path/../swipl/lib/x86_64-darwin15.6.0/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	@executable_path/../swipl/lib/x86_64-darwin15.6.0/libreadline.6.dylib (compatibility version 6.0.0, current version 6.3.0)
```

The last is a copy of GNU readline as compiled from MacPorts running MacOS 10.11.6 (El Captain).
I don't think the MacOS version matters for the Macport binaries.  Just updated Macports to the latest version.   I doubt it will help, but one never knows.   If anyone sorts out what version of libreadline works with what version of Terminal, I'm happy to use that version, but I don't feel responsible for this.

**Comment by sandiway** [comment]
Hi,

I installed the latest version of MacPorts for macOS Sierra (10.12)
I then did 
sudo port install swi-prolog
swipl
Welcome to SWI-Prolog (Multi-threaded, 64 bits, Version 7.2.3)

I can confirm the Delete key on my MacBook works fine on 7.2.3 in Terminal on macOS Sierra.
However, I thought the SWI-Prolog documentation said to install Version 7.3.x (latest 7.3.28)
So:
sudo port uninstall swi-prolog
--->  Deactivating swi-prolog @7.2.3_1+mt
--->  Cleaning swi-prolog
--->  Uninstalling swi-prolog @7.2.3_1+mt
--->  Cleaning swi-prolog
sudo port install swi-prolog-devel
which swipl
/opt/local/bin/swipl
swipl
Welcome to SWI-Prolog (Multi-threaded, 64 bits, Version 7.3.28)
And the Delete key works fine too!

And:
/Applications/SWI-Prolog.app/Contents/MacOS/swipl
Welcome to SWI-Prolog (Multi-threaded, 64 bits, Version 7.3.27)
The damn Delete key now works fine here too!
